cmd/pulumi-test-language/tests.go

```diff
@@ -1163,6 +1163,57 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	"l2-invoke-options-depends-on": {
+		providers: []plugin.Provider{&providers.SimpleInvokeProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, err, changes)
+					require.Len(l, snap.Resources, 5, "expected 5 resources")
+					// TODO: the root stack must be the first resource to be registered
+					// such that snap.Resources[0].Type == resource.RootStackType
+					// however with the python SDK, that is not the case, instead the default
+					// provider gets registered first. This is indicating that something might be wrong
+					// with the how python SDK registers resources
+					var stack *resource.State
+					for _, r := range snap.Resources {
+						if r.Type == resource.RootStackType {
+							stack = r
+							break
+						}
+					}
+
+					require.NotNil(l, stack, "expected a stack resource")
+					outputs := stack.Outputs
+					assertPropertyMapMember(l, outputs, "hello", resource.NewStringProperty("hello world"))
+
+					var first *resource.State
+					var second *resource.State
+					for _, r := range snap.Resources {
+						if r.URN.Name() == "first" {
+							first = r
+						}
+						if r.URN.Name() == "second" {
+							second = r
+						}
+					}
```

Comment:
We could combine these loops perhaps.

Reply:
I prefer to leave the loop for the root stack separate, as it mirrors the other tests.